### DISCUSSION TITLE
已经将paopaodns指向泡泡网关，如何使局域网中其他电脑IP（IPV4）只使用paopaodns，不走泡泡网关

### DISCUSSION THREAD

**Discussion by gukey** [kkkgo/PaoPaoDNS#47] [问答和帮助]
如题，因为有xbox游戏机，如果将默认dns设置为paopaodns的服务器地址，而paopaodns和泡泡网关是联动的，这样子就会让主机更新的部分流量经过泡泡网关，造成流量的浪费，目前的方法是只把主要电脑使用paopaodns，来科学上网，其他默认使用114 的dns（通过ROS的DHCP 分配dns），但是又觉得有现成的dns服务器不用，有点浪费，想请教下，能否有可用的方法，就是只让局域网内的电脑使用dns功能，但流量不经过clash网关；
[![image](https://github.com/kkkgo/PaoPaoDNS/assets/16250885/22693725-a062-4730-bf18-ea8f33e7c55f)](http://forumupload.ru/uploads/001b/b0/1b/8/770664.png)
![](http://forumupload.ru/uploads/001b/b0/1b/8/901583.png)

**Comment by kkkgo** [comment]
因为53端口只有一个，所以结果自然也是唯一的，你可以多开一个docker作备用即可。

**Reply by gukey** [reply]
能详细说说吗，是在同一个虚拟机在开一个docker，但53端口不是已经占用了吗，还是需要重新在虚拟机架一个。

**Reply by kkkgo** [reply]
你可以使用macvlan，开无限个都行，macvlan就是独立IP。

**Reply by gukey** [reply]
哦，理解了

**Comment by kkkgo** [comment]
或者你可以把xbox的下载域名加入list。在下载的时候你可以在web界面观察到下载流量和域名。

**Reply by gukey** [reply]
怎么在web界面查看流量？

**Reply by kkkgo** [reply]
![image](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/6021b761-7770-48e2-ba70-9235cbe2446d)

**Reply by gukey** [reply]
好的，我试试

**Reply by gukey** [reply]
感谢回复；

**Reply by kkkgo** [reply]
steam游戏下载跳过list：
force_nocn_list.txt 添加：
```
domain:steamcontent.com
domain:akamaized.net
domain:akamaihd.net
```
欢迎分享你的规则。

**Reply by kkkgo** [reply]
参考https://github.com/skydevil88/XboxDownload ，
可以使用规则跳过：
```
domain:xboxlive.com
```

**Comment by kkkgo** [comment]
![1696683187943](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/d103cdde-1755-4243-9ac7-007f1ebcbcca)

**Comment by kkkgo** [comment]
PPGW新增网络流量记录功能，详情查看最新release。
https://github.com/kkkgo/PaoPaoGateWay/releases

- 新增net_rec选项
# net_rec网络流量记录功能
`net_rec`选项：网络流量记录功能，可以记录网关连接了哪些域名、上传下载消耗了多少流量、客户端IP，并默认按照消耗流量的大小排序，该功能可以根据实际情况方便地调整分析自己的域名规则列表。设置为`yes`开启该功能后可以在web界面点击[记录]，选择下载表格或者在线加载。注意事项：
  - 1、 重启或者修改ppgw的密码、Web端口、关了再开`net_rec`选项会导致数据清空。
  - 2、如果流量太小连接持续时间过短，有可能在记录之前连接已经关闭，流量会显示为0B。
  - 3、客户端IP记录的是最后一次连接该域名的客户端IP。
  - 4、数据下载没有身份验证。
  - 5、理论上会略微增加资源占用，取决于你的并发连接数量，可适当增加运行资源。   
  
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/4530897/52f4991f-761e-4120-94bc-695d36813a32)

**Reply by jq88kazze** [reply]
paopaodns+gw的组合，今天试了一下地平线5连不上服务器，用223.5.5.5就可以连上

**Reply by kkkgo** [reply]
排查节点问题：
把网关和DNS均设置为ppgw，如果可以连上，则节点无问题。如果不能连上，说明节点不能连上服务器。

**Reply by kkkgo** [reply]
如果希望游戏相关域名跳过PPGW，可以使用同样的方法结合net_rec功能抓出对应的域名，可以添加到`force_nocn_list.txt`

**Comment by TwoBall** [comment]
steam还要加上steamserver.net，连不上国内的cdn下载速度贼慢

**Reply by TwoBall** [reply]
参考https://www.dogfight360.com/blog/knowledge-base/fix_steamdl_region/

**Reply by kkkgo** [reply]
steam可以在设置里面手动选择广州服务器。

**Reply by Edmondguo** [reply]
确实哎，不加domain:steamserver.net，steam控制台显示是hk，clash里看虽然没有走代理，但是只有10-20MB/s，加了之后steam控制台显示是cn，下载速度可以跑满500M带宽达到60MB/s了
